### PR TITLE
impl(internal/gitrepo): add functionality to clone .git repos

### PR DESCRIPTION
**Pull request by julieqiu**
Update flags. Add functionality to open and clone Git repositories, and update the generator logic to clone `googleapis` and `google-cloud-<language>`.

### CODE REVIEW

File: internal/gitrepo/gitrepo.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+}
+
+// CloneOrOpen provides access to a Git repository.
+//
+// If a repository already exists at the specified directory path (dirpath),
+// it opens and provides access to that repository.
+//
+// Otherwise, it clones the repository from the given URL (repoURL) and saves it
+// to the specified directory path (dirpath).
+func CloneOrOpen(ctx context.Context, dirpath, repoURL string) (*Repo, error) {
```
Comment (jskeet) [inline internal/gitrepo/gitrepo.go:41]: At some point we're likely to want a branch (or maybe committish) but that can come later.

File: internal/generate/run.go [hunk 2]
```diff
@@ -16,13 +16,37 @@ package generate
...
+const googleapisURL = "https://github.com/googleapis/googleapis"
+
+func cloneGoogleapis(ctx context.Context) (*gitrepo.Repo, error) {
+	repoPath := filepath.Join(os.TempDir(), "/generator-googleapis")
+	return gitrepo.CloneOrOpen(ctx, repoPath, googleapisURL)
+}
+
+func cloneLanguageRepo(ctx context.Context, language string) (*gitrepo.Repo, error) {
+	languageRepoURL := fmt.Sprintf("https://github.com/googleapis/google-cloud-%s", language)
+	repoPath := filepath.Join(os.TempDir(), fmt.Sprintf("/google-cloud-%s", language))
```
Comment (jskeet) [inline internal/generate/run.go:50]: Maybe `/generator-google-cloud-%s` to be consistent with `generator-googleapis`?

**Review by jskeet — APPROVED** [review]
One potential change and a note for the future, but can be merged as-is.

File: internal/generate/flags.go [hunk 3]
```diff
@@ -62,11 +60,15 @@ Flags:
...
+	if cfg.api == "" {
+		return fmt.Errorf("api must be provided")
+	}
+
 	switch cfg.language {
 	case "cpp":
 		return errNotImplemented
-	case "csharp":
-		return errNotImplemented
+	case "dotnet":
```
Comment (jskeet) [inline internal/generate/flags.go:70]: We need to modify the help text for this - and remove the "command" part, as that's not currently used as far as I can see (and prevented the flags from being parsed).

File: internal/generate/run.go (hunk 2 above)
Comment (julieqiu) [reply internal/generate/run.go:50]: Done

File: internal/generate/flags.go (hunk 3 above)
Comment (julieqiu) [reply internal/generate/flags.go:70]: Done

**Pull request merged**